File: Data Structures & Algorithms/string-encode-and-decode/submission-4.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string res;
        for(int i = 0; i < strs.size(); i++)
        {
            res+=strs[i];
            res+='\n';
        }
        return res;
    }

    vector<string> decode(string s) {
        vector<string> res;
        string curr;
        int i = 0;
        while(i < s.size())
        {
            if(s[i] == '\n')
            {
                res.push_back(curr);
                curr = "";
            }
            else
            {
                curr+=s[i];
            }
            i++;
        }
        // res.push_back(curr);
        return res;
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-4.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        res = 1
        neg = False
        if n < 0:
            neg = True
            n *= -1
        while n > 0:
            if n%2 == 1:
                res *= x
            n = n // 2
            x *= x
        if neg:
            return 1/res
        return res